contracts/yellow-button/lib.rs

```diff
@@ -0,0 +1,545 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use ink_lang as ink;
+
+/// This is the YellowButton
+/// Rewards are distributed for extending the life of the button for as long as possible:
+/// user_score = deadline - now
+/// Pressiah gets 50% of tokens
+/// the game is played until TheButton dies
+
+#[ink::contract]
+mod yellow_button {
+
+    use button_token::{BALANCE_OF_SELECTOR, TRANSFER_SELECTOR};
+    use ink_env::{
+        call::{build_call, Call, ExecutionInput, Selector},
+        DefaultEnvironment, Error as InkEnvError,
+    };
+    use ink_lang::{codegen::EmitEvent, reflect::ContractEventBase};
+    use ink_prelude::{borrow::ToOwned, string::String, vec::Vec};
+    use ink_storage::{traits::SpreadAllocate, Mapping};
+
+    /// Error types
+    #[derive(Debug, PartialEq, Eq, scale::Encode, scale::Decode)]
+    #[cfg_attr(feature = "std", derive(scale_info::TypeInfo))]
+    pub enum Error {
+        /// Returned if given account already pressed The Button
+        AlreadyParticipated,
+        /// Returned if button is pressed after the deadline
+        AfterDeadline,
+        /// Account not whitelisted to play
+        NotWhitelisted,
+        /// Returned when an account which is not the owner calls a method with access control
+        NotOwner,
+        /// Returned if a call to another contract has failed
+        ContractCall(String),
```

Comment:
How about `ContractCall(InkEnvError)`? That way you wouldn't need to manually do the conversion below.

Reply:
won;t work AFAIK: [ink_env::Error](https://docs.rs/ink_env/latest/ink_env/enum.Error.html) is not Encode